File: first_page_module/src/main/java/com/phone/first_page_module/ui/VideoListActivity.java
package com.phone.first_page_module.ui;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.FrameLayout;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.Toolbar;
import androidx.recyclerview.widget.DefaultItemAnimator;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.alibaba.fastjson.JSONObject;
import com.phone.common_library.base.BaseMvpRxAppActivity;
import com.phone.common_library.base.IBaseView;
import com.phone.common_library.manager.LogManager;
import com.phone.common_library.manager.ResourcesManager;
import com.phone.first_page_module.R;
import com.phone.first_page_module.adapter.VideoListAdapter;
import com.phone.first_page_module.bean.VideoListBean;
import com.phone.first_page_module.presenter.FirstPagePresenterImpl;
import com.phone.first_page_module.view.IVideoListView;
import com.scwang.smart.refresh.layout.SmartRefreshLayout;

import java.util.ArrayList;
import java.util.List;

/**
 * author    : Urasaki
 * e-mail    : 1164688204@qq.com
 * date      : 2020/11/19 16:04
 * introduce :
 */

public class VideoListActivity extends BaseMvpRxAppActivity<IBaseView, FirstPagePresenterImpl>
        implements IVideoListView {

    private static final String TAG = "VideoListActivity";
    private Toolbar toolbar;
    private FrameLayout layoutBack;
    private ImageView imvBack;
    private TextView tevTitle;
    private SmartRefreshLayout refreshLayout;
    private RecyclerView rcvData;

    private final List<VideoListBean.LargeImageListBean> videoListBeanList = new ArrayList<>();
    //    private boolean isRefresh;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    protected int initLayoutId() {
        return R.layout.activity_video_list;
    }

    @Override
    protected void initData() {
        //        isRefresh = true;
        Intent intent = getIntent();
        Bundle bundle = intent.getExtras();
        String data = bundle.getString("data");
        LogManager.i(TAG, "data*****" + data);
        VideoListBean videoListBean = JSONObject.parseObject(data, VideoListBean.class);
        for (int i = 0; i < 15; i++) {
            videoListBeanList.add(videoListBean.getLarge_image_list().get(0));
        }
    }

    @Override
    protected void initViews() {
        toolbar = (Toolbar) findViewById(R.id.toolbar);
        layoutBack = (FrameLayout) findViewById(R.id.layout_back);
        imvBack = (ImageView) findViewById(R.id.imv_back);
        tevTitle = (TextView) findViewById(R.id.tev_title);
        refreshLayout = (SmartRefreshLayout) findViewById(R.id.refresh_layout);
        rcvData = (RecyclerView) findViewById(R.id.rcv_data);

        setToolbar(false, R.color.color_FFE066FF);
        imvBack.setColorFilter(ResourcesManager.getColor(R.color.color_FFFFFFFF));

        layoutBack.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                finish();
            }
        });
        initAdapter();

        //        Glide.with(this).load("")
        //            .placeholder(com.phone.common_library.R.mipmap.ic_launcher)
        //            .listener(new RequestListener<Drawable>() {
        //
        //                @Override
        //                public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
        //                    return false;
        //                }
        //
        //                @Override
        //                public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
        //                    return false;
        //                }
        //            })
        //            .into(imvBack);
    }

    private void initAdapter() {
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(this);
        linearLayoutManager.setOrientation(RecyclerView.VERTICAL);
        rcvData.setLayoutManager(linearLayoutManager);
        rcvData.setItemAnimator(new DefaultItemAnimator());

        VideoListAdapter videoListAdapter = new VideoListAdapter(this);
        //        videoListAdapter.setRcvOnItemViewClickListener(new RcvOnItemViewClickListener() {
        //            @Override
        //            public void onItemClickListener(int position, View view) {
        //
        //            }
        //        });
        rcvData.setAdapter(videoListAdapter);
        videoListAdapter.clearData();
        videoListAdapter.addData(videoListBeanList);

        refreshLayout.setEnableRefresh(false);
        refreshLayout.setEnableLoadMore(false);
        //        refreshLayout.setOnRefreshLoadMoreListener(new OnRefreshLoadMoreListener() {
        //            @Override
        //            public void onLoadMore(@NonNull RefreshLayout refreshLayout) {
        //                LogManager.i(TAG, "onLoadMore");
        //                isRefresh = false;
        //
        //            }
        //
        //            @Override
        //            public void onRefresh(@NonNull RefreshLayout refreshLayout) {
        //                isRefresh = true;
        //
        //            }
        //        });
    }

    @Override
    protected void initLoadData() {

    }

    @NonNull
    @Override
    protected FirstPagePresenterImpl attachPresenter() {
        return new FirstPagePresenterImpl(this);
    }

    @Override
    public void videoListSuccess(String success) {

    }

    @Override
    public void videoListError(String error) {

    }

    @Override
    public void showLoading() {
        if (getLoadView() != null && !getLoadView().isShown()) {
            getLoadView().setVisibility(View.VISIBLE);
            getLoadView().start();
        }
    }

    @Override
    public void hideLoading() {
        if (getLoadView() != null && getLoadView().isShown()) {
            getLoadView().stop();
            getLoadView().setVisibility(View.GONE);
        }
    }

}

File: module_home/src/main/java/com/phone/module_home/ui/VideoListActivity.kt
package com.phone.module_home.ui

import android.view.View
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.DefaultItemAnimator
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.alibaba.fastjson.JSONObject
import com.phone.library_common.base.BaseMvpRxAppActivity
import com.phone.library_common.base.IBaseView
import com.phone.library_common.manager.LogManager.i
import com.phone.library_common.manager.ResourcesManager
import com.phone.module_home.R
import com.phone.module_home.adapter.LargeImageListBean
import com.phone.module_home.adapter.VideoListAdapter
import com.phone.module_home.adapter.VideoListBean
import com.phone.module_home.presenter.HomePresenterImpl
import com.phone.module_home.view.IVideoListView
import com.scwang.smart.refresh.layout.SmartRefreshLayout

class VideoListActivity : BaseMvpRxAppActivity<IBaseView, HomePresenterImpl>(),
    IVideoListView {

    companion object {
        @JvmStatic
        private val TAG = VideoListActivity::class.java.simpleName
    }

    private var toolbar: Toolbar? = null
    private var layoutBack: FrameLayout? = null
    private var imvBack: ImageView? = null
    private var tevTitle: TextView? = null
    private var refreshLayout: SmartRefreshLayout? = null
    private var rcvData: RecyclerView? = null

    private val videoListBeanList: MutableList<LargeImageListBean> = ArrayList()
    //    private boolean isRefresh;

    override fun initLayoutId(): Int {
        return R.layout.home_activity_video_list
    }

    override fun initData() {
        //        isRefresh = true;
        val intent = intent
        val bundle = intent.extras
        val data = bundle?.getString("data")
        i(TAG, "data*****$data")
        val videoListBean = JSONObject.parseObject(
            data,
            VideoListBean::class.java
        )
        for (i in 0..14) {
            videoListBeanList.add(videoListBean.large_image_list[0])
        }
    }

    override fun initViews() {
        toolbar = findViewById<View>(R.id.toolbar) as Toolbar
        layoutBack = findViewById<View>(R.id.layout_back) as FrameLayout
        imvBack = findViewById<View>(R.id.imv_back) as ImageView
        tevTitle = findViewById<View>(R.id.tev_title) as TextView
        refreshLayout = findViewById<View>(R.id.refresh_layout) as SmartRefreshLayout
        rcvData = findViewById<View>(R.id.rcv_data) as RecyclerView
        setToolbar(false, R.color.library_color_FFE066FF)
        imvBack?.setColorFilter(ResourcesManager.getColor(R.color.library_color_FFFFFFFF))
        layoutBack?.setOnClickListener { finish() }
        initAdapter()

        //        Glide.with(this).load("")
        //            .placeholder(com.phone.common_library.R.mipmap.ic_launcher)
        //            .listener(new RequestListener<Drawable>() {
        //
        //                @Override
        //                public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<Drawable> target, boolean isFirstResource) {
        //                    return false;
        //                }
        //
        //                @Override
        //                public boolean onResourceReady(Drawable resource, Object model, Target<Drawable> target, DataSource dataSource, boolean isFirstResource) {
        //                    return false;
        //                }
        //            })
        //            .into(imvBack);
    }

    private fun initAdapter() {
        val linearLayoutManager = LinearLayoutManager(this)
        linearLayoutManager.orientation = RecyclerView.VERTICAL
        rcvData?.layoutManager = linearLayoutManager
        rcvData?.itemAnimator = DefaultItemAnimator()
        val videoListAdapter = VideoListAdapter(this)
        //        videoListAdapter.setOnItemViewClickListener(new RcvOnItemViewClickListener() {
        //            @Override
        //            public void onItemClickListener(int position, View view) {
        //
        //            }
        //        });
        rcvData?.adapter = videoListAdapter
        videoListAdapter.clearData()
        videoListAdapter.addData(videoListBeanList)
        refreshLayout?.setEnableRefresh(false)
        refreshLayout?.setEnableLoadMore(false)
        //        refreshLayout.setOnRefreshLoadMoreListener(new OnRefreshLoadMoreListener() {
        //            @Override
        //            public void onLoadMore(@NonNull RefreshLayout refreshLayout) {
        //                LogManager.i(TAG, "onLoadMore");
        //                isRefresh = false;
        //
        //            }
        //
        //            @Override
        //            public void onRefresh(@NonNull RefreshLayout refreshLayout) {
        //                isRefresh = true;
        //
        //            }
        //        });
    }

    override fun initLoadData() {}

    override fun attachPresenter(): HomePresenterImpl {
        return HomePresenterImpl(this)
    }

    override fun videoListSuccess(success: String) {}

    override fun videoListError(error: String) {}

    override fun showLoading() {
        if (!mLoadView.isShown) {
            mLoadView.visibility = View.VISIBLE
            mLoadView.start()
        }
    }

    override fun hideLoading() {
        if (mLoadView.isShown) {
            mLoadView.stop()
            mLoadView.visibility = View.GONE
        }
    }

}